terragrunt/modules/crates-io/s3-static.tf

```diff
@@ -27,6 +27,17 @@ resource "aws_s3_bucket" "static" {
     }
   }
 
+  lifecycle_rule {
+    id      = "purge-db-dump-zip"
+    enabled = true
+    prefix  = "db-dump.zip"
+
+    abort_incomplete_multipart_upload_days = 1
+    noncurrent_version_expiration {
+      days = 1
+    }
+  }
```

Comment:
I'd like to avoid the duplication here. Can we use a [dynamic block](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks) for this instead?

```hcl
dynamic "lifecycle_rule" {
  for_each = toset(["db-dump.tar.gz", "db-dump.zip"])

  content {
    id      = "purge-${replace(lifecycle_rule.key, ".", "-")}"
    enabled = true
    prefix  = lifecycle_rule.key

    abort_incomplete_multipart_upload_days = 1
    noncurrent_version_expiration {
      days = 1
    }
  }
}
```

Reply:
> Can we use a [dynamic block](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks) for this instead?

you tell me, I have no idea how to check if this works or not 😂

Reply:
Sorry, let me rephrase the question. Can you please change it to a dynamic block? 😂 I already tested that the code above works.

Reply:
done :)